ee/query-service/app/api/gateway.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+import (
+	"net/http"
+	"strings"
+
+	"go.signoz.io/signoz/ee/query-service/integrations/gateway"
+)
+
+func (ah *APIHandler) ServeGatewayHTTP(rw http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if !strings.HasPrefix(req.URL.Path, gateway.RoutePrefix+gateway.AllowedPrefix) {
+		rw.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	license, err := ah.LM().GetRepo().GetActiveLicense(ctx)
+	if err != nil {
+		RespondError(rw, err, nil)
+		return
+	}
+
+	//Create headers
+	var licenseKey string
+	if license != nil {
+		licenseKey = license.Key
+	}
+
+	req.Header.Set("X-Signoz-Cloud-Api-Key", licenseKey)
+	req.Header.Set("X-Consumer-Username", "lid:00000000-0000-0000-0000-000000000000")
+	req.Header.Set("X-Consumer-Groups", "ns:default")
+
+	ah.Gateway().ServeHTTP(rw, req)
```

Comment:
Does this panic with nil pointer dereference if the gateway URL is not provided? I see the feature will be inactive but seems like it would crash if the gateway URL is invoked.

Reply:
Yeah it will panic. Not ideal I know. Do you want me to workaround it?

I'll have to create an interface in noop to override the serveHTTP behaviour.

Reply:
I do not expect users/frontend to hit this URL when the gateway is not enabled but it's a good idea to not panic because of nil pointer dereference. You can address it later if you perfer.

Reply:
Got it!